src/components/PersonalSettings.vue

```diff
@@ -147,9 +165,46 @@ export default {
 			documentSigningCert: this.initial.documentSigningCert || '',
 			documentSigningKey: this.initial.documentSigningKey || '',
 			documentSigningCa: this.initial.documentSigningCa || '',
+			tokenGenerated: false,
+			accessToken: '',
+			accessTokenTTL: '',
+			userId: '',
+			wopiSettingBaseUrl: '',
+			public_wopi_url: this.initial.publicWopiUrl || '',
+		}
+	},
+	computed: {
+		shareToken() {
+			return getSharingToken()
+		},
+	},
+	async mounted() {
+		const currentUser = getCurrentUser()
+		if (currentUser && currentUser.uid) {
+			this.userId = currentUser.uid
+			await this.generateAccessToken()
+			if (this.accessToken) {
+				this.wopiSettingBaseUrl = getConfigFileUrl()
+				console.debug('wopiSettingBaseUrl', this.wopiSettingBaseUrl)
+				this.tokenGenerated = true
+			}
+		} else {
+			console.error('User not authenticated')
 		}
 	},
 	methods: {
+		async generateAccessToken() {
+			const { data } = await axios.get(generateUrl('/apps/richdocuments/settings/generateToken/user'))
+			if (data.token) {
+				this.accessToken = data.token
+				this.accessTokenTTL = data.token_ttl
+				console.debug('Admin settings WOPI token generated:', this.accessToken, this.accessTokenTTL)
+			} else if (data.federatedUrl) {
+				console.error('Federated URL returned, not expected for admin settings.')
```

Comment:
I don't think this case needs handling, we rather can take the default else branch there.

Reply:
Done